lib/Sema/CSSimplify.cpp

```diff
@@ -2066,6 +2066,14 @@ repairFailures(ConstraintSystem &cs, Type lhs, Type rhs,
   }
 
   case ConstraintLocator::ContextualType: {
+    // If both types are key path, the only differences
+    // between them are mutability and/or root, value type mismatch.
+    if (isKnownKeyPathType(lhs) && isKnownKeyPathType(rhs)) {
```

Comment:
Would it make sense to generalize this repair beyond key paths? When the contextual type is a class, we might *always* see if any of its superclasses would typecheck—"`makeView()` returns a `UIView` but the context requires a `UITableView`.

Reply:
Yes, my plan is to refactor `diagnoseContextualFailures` from `CSDiag` into `ContextualMismatch` failures, I just wanted to make this special case for key path right now because they are particularly bad at the moment.